thunder/core/transforms.py

```diff
@@ -1357,6 +1357,28 @@ def _embedding_prim_grad(
 
 register_grad(pids.EMBEDDING, _embedding_prim_grad)
 
+
+def _maximum_grad(a: TensorProxy, b: TensorProxy, /):
+    fwd = prims.maximum(a, b)
+
+    g = get_grad(fwd)
+
+    # Compute sub-gradient if `a == b`
```

Comment:
Elaborating a little on this comment might be nice -- about how we want to split the grad in this case

Reply:
What happens if an element in `a` or `b` is `NaN`? What happens if the same element of `a` and `b` is a `NaN`?

Reply:
As `NaN > anything = False` and `NaN == NaN = False` , so even if one element happens to be `NaN`, gradient will propate to both elements

```python
sub_g = where(NaN == 42 i.e. False, g / 2, g)  # sub_grad = g

grad_a = where(NaN > 42 i.e. False, g, sub_g)  # grad_a = sub_g = g
grad_b = where(42 > NaN i.e. False, g, sub_g)  # grad_b = sub_g = g
```

Reply:
Great! Let's update the comment to explain these cases

Reply:
Have updated the comment and addressed other comments. This should be ready for another round. Thanks